enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,337 @@
+  interactive installation.
+- Require no machines in the cluster environment other than those that are to
+  become part of the cluster.
+- Install clusters with single-node, compact, and high-availability topologies.
+- Be agnostic to the tools used to provision (virtual) machines so that users
+  can use their own (possibly pre-existing) tooling for provisioning.
+- Support install configurations with platform `none`, `baremetal`, or
+  `vsphere`.
+- Run on either Linux (any distro) or MacOS, in either x86-64 or arm64
+  architectures.
+- Install to machines with either x86-64 or arm64 architectures, independent of
+  the architecture the CLI tool runs on.
+
+### Non-Goals
+
+- Replace the IPI installer in any capacity.
+- Automate booting of the image on (virtual) machines.
+- Support install configurations for cloud-based platforms.
+- Provide an interactive API to drive automated deployments.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
+- Provide a mechanism to add and remove nodes from the cluster after it is
+  installed.
+- Generate image formats other than ISO.
+- Create minimal ISO images with a separate rootfs.
+- Run on Windows.
+
+## Proposal
+
+A command-line tool will enable users to build a custom RHCOS image, initially
+in ISO format, containing the components needed to install.
+
+A prerequisite for disconnected users is that the release image, and possibly
+an extra image containing installer components, is mirrored to a local registry
+that is accessible from the network environment where the cluster is to be
+deployed. The command-line tool will
+[retrieve](https://github.com/openshift/machine-os-images/) the base ISO from
+the container registry, and add an Ignition file to produce a cluster-specific
+install ISO. Users can then boot this image on the hosts that are to become
+part of the cluster using any method they choose - this could be existing
+automation that they have or manually booting using physical or virtual media.
+Container images required for installation will be pulled from the same
+registry as the base ISO, as will the OpenShift release that is being
+installed. One host (hereafter referred to as "Node 0") will be selected to run
+the assisted-service. All hosts will run the assisted agent. A single ISO will
+be used for all hosts in the cluster.
+
+In the automated workflow, all configuration data for the cluster to be
+installed (including the expected number of hosts) will be provided as input to
+the command-line tool at the time the ISO is generated. Users will (optionally)
+be able to supply custom manifests to augment the initial configuration, in
+addition to data required for installation such as the network config.
+Installation will proceed automatically once the requisite hosts have
+registered with Node 0 and passed validations. An API running on Node 0 will
```

Comment:
Is the API the assisted-service REST-API?

Reply:
Yes, that's right.

Reply:
Probably, but the vagueness was intentional (we may rule out e.g. a k8s API running on node0 and started prior to bootstrapping, but not in this enhancement).